bot/cogs/snekbox.py

```diff
@@ -57,39 +42,38 @@
 
 class Snekbox:
     """
-    Safe evaluation using Snekbox
+    Safe evaluation of Python code using Snekbox
     """
 
     def __init__(self, bot: Bot):
         self.bot = bot
         self.jobs = {}
 
-    @property
-    def rmq(self) -> RMQ:
-        return self.bot.get_cog("RMQ")
-
-    @command(name='eval', aliases=('e',))
-    @guild_only()
-    @in_channel(Channels.bot, bypass_roles=BYPASS_ROLES)
-    async def eval_command(self, ctx: Context, *, code: str = None):
-        """
-        Run some code. get the result back. We've done our best to make this safe, but do let us know if you
-        manage to find an issue with it!
-
-        This command supports multiple lines of code, including code wrapped inside a formatted code block.
-        """
+    async def post_eval(self, code: str) -> dict:
+        """Send a POST request to the Snekbox API to evaluate code and return the results."""
+        url = URLs.snekbox_eval_api
+        data = {"input": code}
+        async with self.bot.http_session.post(url, json=data, raise_for_status=True) as resp:
+            return await resp.json()
 
-        if ctx.author.id in self.jobs:
-            await ctx.send(f"{ctx.author.mention} You've already got a job running - please wait for it to finish!")
-            return
+    async def upload_output(self, output: str) -> Optional[str]:
+        """Upload the eval output to a paste service and return a URL to it if successful."""
+        log.trace("Uploading full output to paste service...")
 
-        if not code:  # None or empty string
-            return await ctx.invoke(self.bot.get_command("help"), "eval")
+        url = URLs.paste_service.format(key="documents")
+        try:
+            async with self.bot.http_session.post(url, data=output, raise_for_status=True) as resp:
+                data = await resp.json()
 
-        log.info(f"Received code from {ctx.author.name}#{ctx.author.discriminator} for evaluation:\n{code}")
-        self.jobs[ctx.author.id] = datetime.datetime.now()
+            if "key" in data:
+                return URLs.paste_service.format(key=data["key"])
+        except Exception:
+            # 400 (Bad Request) means the data is too large
```

Comment:
I think this actually happens due to too many characters. More specifically, I believe I pinpointed it to 100k characters. Can this be confirmed?

A check can be added for the output length. Either uploading can be skipped or the output can be truncated to 100k characters.